src/main/scala/com/mozilla/telemetry/streaming/EventsToAmplitude.scala

```diff
@@ -160,7 +166,9 @@ object EventsToAmplitude extends StreamingJobBase {
         try {
           parsePing(Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]]), sample, config)
         } catch {
-          case _: Throwable if !raiseOnError => Array[String]()
+          case NonFatal(ex) if !raiseOnError =>
+            log.error("Encountered an error!", ex)
+            Array[String]()
```

Comment:
I now see that we have this pattern of catching any throwable in several places, so maybe I shouldn't change this. It seems super dangerous, but maybe this is an accepted pattern in Spark?

Reply:
I think @akkomar had some opinions about this -- but the main consideration I believe was to keep streaming jobs running